db/python/tables/cohort.py

```diff
@@ -215,14 +286,47 @@ async def get_cohort_by_id(self, cohort_id: int) -> CohortInternal:
         """
         Get the cohort by its ID
         """
+
         _query = """
-        SELECT id, name, template_id, author, description, project, timestamp
-        FROM cohort
-        WHERE id = :cohort_id
+        SELECT id as c_id, name as c_name, template_id as c_template_id, author as c_author,
+        description as c_description, project as c_project, timestamp as c_timestamp
+        FROM cohort WHERE id = :cohort_id
         """
 
         cohort = await self.connection.fetch_one(_query, {'cohort_id': cohort_id})
         if not cohort:
             raise ValueError(f'Cohort with ID {cohort_id} not found')
 
+        # status criteria not computed in this function as current usage only consume template id
         return CohortInternal.from_db(dict(cohort))
+
+    async def update_cohort_given_id(
+        self, name: str, description: str, status: CohortStatus, cohort_id: int
+    ):
+        """
+        Update the cohort given its ID
+        """
+
+        # The following fields are allowed to update
+        cohort_fields = {
+            'name': name,
+            'description': description,
+            'status': status.value.upper() if status else None,
+        }
+
+        query_params: dict[str, Any] = {
+            **{k: v for k, v in cohort_fields.items() if v is not None}
```

Comment:
Does this need to do dict unpacking? Would `query_params: dict[str, Any] = {k: v for k, v in cohort_fields.items() if v is not None}` work?

Reply:
addressed in https://github.com/populationgenomics/metamist/pull/1142/commits/e9279f7969fee6cb943518805ee726c3ece2ec80